### PR TITLE
Issue#54: News CRUD

### PR DESCRIPTION
**Pull request by isvyry**
## ZenHab

* [Main ZenHab ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009/issues/ita-social-projects/home/54)

## Code reviewers

- [x] @alexorlenko
- [x] @Arthurpanda  
- [x] @mykytam 
- [x] @omelchenkotimur 
- [ ] @Prosperro-de 
- [x] @reeztl 

### Second Level Review

- [ ] @DzigMS 
- [ ] @kh0ma 

## Summary of issue

Add News' CRUD operations

## Summary of change

As user I want to view news
As admin I want to be provided with all existing news operations

## Testing approach

---

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        ReadNewsDto readNewsDto = newsService.getById(id);
+        ReadNews newsApiResponse = convertToReadNews(readNewsDto);
+        return Response.ok().entity(newsApiResponse).build();
+    }
+
+    @Override
+    public Response updateNews(Long id, UpdateNews updateNews, SecurityContext securityContext) {
+        CreateOrUpdateNewsDto updateNewsDto = CreateOrUpdateNewsDto.builder()
+                .title(updateNews.getTitle())
+                .dateTime(LocalDateTime.now())
```
Comment (kh0ma) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: ???

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Override
+    public Response addNews(CreateNews createNews, SecurityContext securityContext) {
+        CreateOrUpdateNewsDto createNewsDto = CreateOrUpdateNewsDto.builder()
+                .title(createNews.getTitle())
+                .description(createNews.getDescription())
+                .text(createNews.getText())
+                .photoUrl(createNews.getPhotoUrl())
+                .source(createNews.getSource())
+                .dateTime(LocalDateTime.now())
```
Comment (kh0ma) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: ???

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                .build();
+
+        News response = convertToNewsView(newsService.update(id, updateNewsDto));
+        return Response.ok().entity(response).build();
+    }
+
+    private ReadNews convertToReadNews(ReadNewsDto readNewsDto) {
+        ReadNews toResponse = new ReadNews();
+        toResponse.setId(readNewsDto.getId());
+        toResponse.setCreateOrUpdateDate(OffsetDateTime.of(readNewsDto.getCreatedAt(), ZoneOffset.UTC));
```
Comment (kh0ma) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: Please move the timezone to constant.

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/services/impl/NewsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        newsToUpdate.setDescription(updateNewsServiceDto.getDescription());
+        newsToUpdate.setPhotoUrl(updateNewsServiceDto.getPhotoUrl());
+        newsToUpdate.setSource(updateNewsServiceDto.getSource());
+        newsRepository.save(newsToUpdate);
+
+        return ConvertToGetNewsServiceDto(newsToUpdate);
+    }
+
+    @Override
+    public Collection<ReadNewsDto> getAll(Integer pageNumber, Integer pageSize) {
```
Comment (kh0ma) [inline home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/services/impl/NewsServiceImpl.java]: I think we need to return a page but not a collection.

**Review by kh0ma — DISMISSED** [review]

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java (hunk 2 above)
Comment (isvyry) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: What exactly should be fixed?

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java (hunk 3 above)
Comment (isvyry) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: ok, noted, but what package should constants be in?

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/services/impl/NewsServiceImpl.java (hunk 4 above)
Comment (isvyry) [reply home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/services/impl/NewsServiceImpl.java]: so the service layer should return Page, and the api-service provides managing of these pages?

**Review by mykytam — APPROVED** [review]
Seems everything is good 👍

**Review by alexorlenko — APPROVED** [review]

**Review by omelchenkotimur — APPROVED** [review]

**Review by Arthurpanda — APPROVED** [review]

**Pull request merged**